pkg/api/validate/dynamic/dynamic.go

```diff
@@ -287,11 +287,24 @@ func (dv *dynamic) validateCIDRRanges(ctx context.Context, subnets []Subnet, add
 			return err
 		}
 
-		_, net, err := net.ParseCIDR(*s.AddressPrefix)
-		if err != nil {
-			return err
+		// Since AddressPrefix seems to be deprecated for addressPrefixes we need this to validate the CIDR of the new
```

Comment:
Thank you for clearly documenting the intention and marking the TODO. I highly appreciate the clarity here.